### ISSUE TITLE
DBFILESRouteobjects.m has major threading bug

### ISSUE THREAD

**Issue by tomandersen** [dropbox/dropbox-sdk-obj-c#216]
Easily fixed.

There are 50 objects made with no thought about threading. They can get called in threads which will crash the application. (Run with Zombies for the full flavour).

50 instances of code like
  if (!DBFILESAlphaGetMetadata) {
    DBFILESAlphaGetMetadata = [[DBRoute alloc] init:@"alpha/get_metadata"

Needs to be changed to 'dispatch_once'

As a workaround, one could upon launch of their program 'simply' call all 50 static calls in that file in the main thread before doing anything with the dropbox sdk. (there may be other places with the problem). 

I have recorded three or so crashes all attributable to this bug soon after launching my app. (which calls about 5 upload calls at once from the main thread, but then those calls use operations on blocks to get things done).

**Comment by tomandersen** [comment]
The bug happens when say [DBFILESRouteObjects DBFILESDCopyBatch] gets called on two threads almost at the same time. The if statement evaluates false TWICE and then the first constructed instance is not assigned to anything and thus ARC removes it at some random point in the future.

**Comment by tomandersen** [comment]
2018-05-01 17:16:13.082682-0400 MYAPP[85765:18801227] *** -[DBRoute retain]: message sent to deallocated instance 0x6070002eaa10

From this code
```
  DBRoute *route = DBFILESRouteObjects.DBFILESUploadSessionStart;
  DBFILESUploadSessionStartArg *arg = [[DBFILESUploadSessionStartArg alloc] initWithClose:close];
  return [self.client requestUpload:route arg:arg inputStream:inputStream];

```

The orriginal call into the API was made on the main thread ( client.filesRoutes batchUploadFiles )

**Comment by greg-db** [comment]
Thanks for the report!

**Comment by greg-db** [comment]
We've released an update that should fix this issue in v6.2.0. Please give that a try and let us know if you're still seeing any issues. Thanks!
